test/conformance/gateway_conformance_test.go

```diff
@@ -63,35 +68,30 @@ func TestGatewayConformance(t *testing.T) {
 		t.Fatalf("unsupported KongRouterFlavor: %s", rf)
 	}
 
-	cSuite, err := suite.NewExperimentalConformanceTestSuite(
-		suite.ExperimentalConformanceOptions{
-			Options: suite.Options{
-				Client:               k8sClient,
-				GatewayClassName:     gatewayClassName,
-				Debug:                true,
-				CleanupBaseResources: !testenv.IsCI(),
-				BaseManifests:        conformanceTestsBaseManifests,
-				SupportedFeatures:    sets.New(supportedFeatures...),
-				SkipTests:            skippedTests,
-			},
-			ConformanceProfiles: sets.New(
-				suite.HTTPConformanceProfileName,
-			),
-			Implementation: conformancev1alpha1.Implementation{
-				Organization: metadata.Organization,
-				Project:      metadata.ProjectName,
-				URL:          metadata.ProjectURL,
-				Version:      metadata.Release,
-				Contact: []string{
-					path.Join(metadata.ProjectURL, "/issues/new/choose"),
-				},
-			},
-		},
+	opts := conformance.DefaultOptions(t)
+	opts.GatewayClassName = gatewayClassName
+	opts.Debug = true
+	opts.CleanupBaseResources = !testenv.IsCI()
+	opts.BaseManifests = conformanceTestsBaseManifests
+	opts.SupportedFeatures = sets.New(supportedFeatures...)
+	opts.SkipTests = skippedTests
+	opts.ConformanceProfiles = sets.New(
+		suite.HTTPConformanceProfileName,
 	)
+	opts.Implementation = conformancev1.Implementation{
+		Organization: metadata.Organization,
+		Project:      metadata.ProjectName,
+		URL:          metadata.ProjectURL,
+		Version:      metadata.Release,
+		Contact: []string{
+			path.Join(metadata.ProjectURL, "/issues/new/choose"),
+		},
+	}
```

Comment:
This is because there was a PR in GWAPI that modified the behavior of setting default values. The relevant discussion can be found here.  https://github.com/kubernetes-sigs/gateway-api/pull/2864#discussion_r1564249514

@mlavacca has fixed it https://github.com/kubernetes-sigs/gateway-api/pull/2979
Once we update to the latest version, we can use the previous implementation method.